### PR TITLE
Miscellaneous tweaks to transport handling and auth tests

### PR DESCRIPTION
**Pull request by SimonWoolf**
Fix a couple bugs in transport handling (fricking mutable arrays), and a few causes of ably-js test instability. (Though mostly still not passing ci due to https://github.com/ably/infrastructure/issues/711 etc).

Also temporarily running against staging in ci, as sandbox is currently buggered.

### CODE REVIEW

File: spec/common/globals/environment.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 define(function(require) {
 	var defaultLogLevel = 2,
 		environment = isBrowser ? (window.__env__ || {}) : process.env,
-		ablyEnvironment = environment.ABLY_ENV || 'sandbox',
+		ablyEnvironment = environment.ABLY_ENV || 'staging',
```
Comment (paddybyers) [inline spec/common/globals/environment.js]: Do we really want to commit this?

File: spec/common/modules/shared_helper.js [hunk 2]
```diff
@@ -90,7 +90,15 @@ define(['spec/common/modules/testapp_module', 'spec/common/modules/client_module
 
 		/* testFn is assumed to be a function of realtimeOptions that returns a nodeunit test */
 		function testOnAllTransports(exports, name, testFn, excludeUpgrade) {
-			utils.arrForEach(availableTransports, function(transport) {
+			/* Don't include jsonp if possible. Why? Because you can't abort requests. So recv's
```
Comment (paddybyers) [inline spec/common/modules/shared_helper.js]: So when will jsonp get tested?

**Comment by paddybyers** [comment]
LGTM apart from a couple of comments.

File: spec/common/globals/environment.js (hunk 1 above)
Comment (SimonWoolf) [inline spec/common/globals/environment.js]: Nope, was just to change the ci run. (Hence the commit message of `TEMP NOTFORMERGING` :) )

File: spec/common/modules/shared_helper.js (hunk 2 above)
Comment (SimonWoolf) [inline spec/common/modules/shared_helper.js]: In `spec/browser/simple.test.js` (same as before I added the `testOnAllTransports` helper)

**Comment by SimonWoolf** [comment]
Merged as 5e6b19a

**Pull request closed without merging**